tests/main/test_model_rebuild_validation.py

```diff
@@ -0,0 +1,59 @@
+"""
+To ensure all models with nested dependencies are fully rebuilt this test will
+create an instance of the query from `multiple_fragments` containing a `FullA`
+(extended from `ExampleQuery2ExampleQuery`) which in turn holds a `FullB`
+(extended from `FullAFieldB`).
+
+If this model is not rebuilt with `FullA.model_rebuild()` `ExampleQuery2` will
+not be fully defined and we will get a `PydanticUserError`.
+
+Reference to Pydantic documentation about when and why we need to call
+`model_rebuild` on our types:
+https://errors.pydantic.dev/2.5/u/class-not-fully-defined
+"""
+
+import pytest
+from pydantic_core import ValidationError
+
+from .clients.multiple_fragments.expected_client.example_query_2 import (
+    ExampleQuery2,
+    ExampleQuery2ExampleQuery,
+)
+from .clients.multiple_fragments.expected_client.fragments import FullA
+
+
+def test_model_rebuild_validate():
```

Comment:
This test reads to me like a four tests bunched together. It should be split into those.

Reply:
Made a fixture and split the test into 4 different steps which also made us able to remove the comments, good idea! 👍